data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -174,6 +174,64 @@ private void overwriteExistingData() throws IOException {
         .commit();
   }
 
+  private void appendData() throws IOException {
```

Comment:
I generally found the tests in TestLocalScan to be a little hard to read. It seems that the sharedTable being used in each test can have a different state depending upon the order of execution of the tests as it is initialized only once before all tests run. Also,  the data records `file1FirstSnapshotRecords` etc are being initialized by each test on their own, maybe we could make them constants. I don't think we need to take action on this though, if u all agree we can take it up as a separate task.

Reply:
Agree with you. The test data starts to become a bit hard to use if we add more tests in future. I could try to clean this up, let me create an issue in case we forget it.